lib/travis/build/script/dart.rb

```diff
@@ -114,10 +114,16 @@ def script
         private
 
           def archive_url
-            if not ["stable", "dev"].include?(config[:dart])
-              sh.failure "Only 'stable' and 'dev' can be used as dart version for now"
+			# support of "dev" or "start"
+			if ["stable", "dev"].include?(config[:dart])
+			  "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}/release/latest"
+			# support of "stable/release/1.15.0" or "be/raw/110749"
+			elsif config[:dart].include?("/")
+			  "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}"
+			# support of "1.15.0" or "1.16.0-dev.2.0"
+			else
+			  "https://storage.googleapis.com/dart-archive/channels/dev/release/#{config[:dart]}"
```

Comment:
Oops, I misread the content from https://www.googleapis.com/storage/v1/b/dart-archive/o?prefix=channels/dev/release/&delimiter=/
I appears that there are not all stable releases on the `dev` channel. For instance I see `1.14.0` but not `1.14.1`.

Finally I should be something like this : 

```ruby
urlEnd = ''
# support of "dev" or "stable"
if ["stable", "dev"].include?(config[:dart])
  urlEnd = "#{config[:dart]}/release/latest"
# support of "stable/release/1.15.0" or "be/raw/110749"
elsif config[:dart].include?("/")
  urlEnd = config[:dart]
# support of dev versions like "1.16.0-dev.2.0" or "1.16.0-dev.2.0"
elsif config[:dart].include?("-dev")
  urlEnd = "dev/release/#{config[:dart]}"
# support of stable versions like "1.14.0" or "1.14.1"
else
  urlEnd = "stable/release/#{config[:dart]}"
end
"https://storage.googleapis.com/dart-archive/channels/#{urlEnd}"
```